types/setting.go

```diff
@@ -895,33 +921,39 @@ func ValidateSetting(name, value string) (err error) {
 
 	case SettingNameStorageOverProvisioningPercentage:
 		if _, err := strconv.Atoi(value); err != nil {
-			return fmt.Errorf("value %v is not a number", value)
+			return errors.Wrapf(err, "value %v is not a number", value)
 		}
 		// additional check whether over provisioning percentage is positive
 		value, err := util.ConvertSize(value)
-		if err != nil || value < 0 {
+		if err != nil {
+			return errors.Wrapf(err, "failed to parse %v as size", value)
+		}
+		if value < 0 {
 			return fmt.Errorf("value %v should be positive", value)
 		}
 	case SettingNameStorageMinimalAvailablePercentage:
 		if _, err := strconv.Atoi(value); err != nil {
-			return fmt.Errorf("value %v is not a number", value)
+			return errors.Wrapf(err, "value %v is not a number", value)
 		}
 		// additional check whether minimal available percentage is between 0 to 100
 		value, err := util.ConvertSize(value)
-		if err != nil || value < 0 || value > 100 {
+		if err != nil {
+			return errors.Wrapf(err, "failed to parse %v as size", value)
+		}
+		if value < 0 || value > 100 {
 			return fmt.Errorf("value %v should between 0 to 100", value)
 		}
 	case SettingNameDefaultReplicaCount:
 		c, err := strconv.Atoi(value)
 		if err != nil {
-			return fmt.Errorf("value %v is not int: %v", SettingNameDefaultReplicaCount, err)
+			return errors.Wrapf(err, "value %v is not number", SettingNameDefaultReplicaCount)
 		}
 		if err := ValidateReplicaCount(c); err != nil {
-			return fmt.Errorf("value %v: %v", c, err)
+			return errors.Wrapf(err, "failed to validate replica count %v", c)
 		}
 	case SettingNameReplicaAutoBalance:
 		if err := ValidateReplicaAutoBalance(longhorn.ReplicaAutoBalance(value)); err != nil {
-			return fmt.Errorf("value %v: %v", value, err)
+			return errors.Wrapf(err, "value %v: %v", value, err)
```

Comment:
```suggestion
			return errors.Wrapf(err, "failed to validate replica auto balance: %v", value)
```

Reply:
Thank you, I will follow the comment to fix.
https://github.com/longhorn/longhorn-manager/blob/de00e06ea1fcd510afee72cb8132231513953f84/types/setting.go#L956